rasa/core/featurizers.py

```diff
@@ -259,6 +287,314 @@ def create_encoded_all_actions(self, domain: Domain) -> np.ndarray:
         return encoded_all_actions
 
 
+class E2ESingleStateFeaturizer(SingleStateFeaturizer):
+    def __init__(self) -> None:
+
+        super().__init__()
+        self.output_shapes = {}
+        self.slot_names = []
+        self.slot_states = []
+        self.form_states = []
+        self.entities = []
+        self.action_names = []
+        self.intents = []
+
+    def prepare_from_domain(self, domain: Domain) -> None:
+        # storing slot names so that the features are always added in the same order
+        self.slot_names = [slot.name for slot in domain.slots]
+        self.slot_states = domain.slot_states
+        self.form_states = domain.form_states
+        self.entities = domain.entities
+        self.action_names = domain.action_names
+        self.intents = domain.intent_states
+
+        self.output_shapes[ENTITIES] = len(
+            self.slot_states + self.form_states + self.entities
+        )
+
+    def _get_slot_and_entity_features(self, state: STATE) -> np.ndarray:
```

Comment:
let's separate extracting slots, form, entity features into different methods

Reply:
Just updated my branch and currently this method is failing as I don't have any entities, slots or forms. We need to add some sanity checks in here.